File: src/main/java/com/gitlab/daring/image/command/feature/StarDetectorCommand.java
package com.gitlab.daring.image.command.feature;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import com.gitlab.daring.image.command.parameter.IntParam;
import org.bytedeco.javacpp.opencv_xfeatures2d.StarDetector;

class StarDetectorCommand extends BaseCommand {

    final IntParam maxSize = intParam(45, "0-100");
    final IntParam th1 = intParam(30, "0-100");
    final IntParam th2 = intParam(10, "0-100");
    final IntParam th3 = intParam(8, "0-100");
    final IntParam nonMaxSize = intParam(5, "0-100");

    public StarDetectorCommand(String... args) {
        super(args);
    }

    @Override
    public void execute(CommandEnv env) {
        env.featureDetector = StarDetector.create(maxSize.v, th1.v, th2.v, th3.v, nonMaxSize.v);
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/feature/StarDetectorCommand.kt
package com.gitlab.daring.image.command.feature

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand
import org.bytedeco.javacpp.opencv_xfeatures2d.StarDetector

internal class StarDetectorCommand(args: Array<String>) : KBaseCommand(args) {

    val maxSize = intParam(45, "0-100")
    val th1 = intParam(30, "0-100")
    val th2 = intParam(10, "0-100")
    val th3 = intParam(8, "0-100")
    val nonMaxSize = intParam(5, "0-100")

    override fun execute(env: CommandEnv) {
        env.featureDetector = StarDetector.create(maxSize.v, th1.v, th2.v, th3.v, nonMaxSize.v)
    }

}